### PR TITLE
[tools/depends][linux] Utilise target_platform for dependency options

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Utilise target_platform for dependency options

## Motivation and context

Build dependencies based on platform type rather than rendersystem. More fine grained dep building for CI.

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
I should also add, im not fussed if we dont go with this, it really doesnt matter in the grand scheme of things.

File: tools/depends/configure.ac [hunk 1]
```diff
@@ -382,7 +382,6 @@ case $host in
     platform_cxxflags="$platform_cflags"
     platform_os="linux"
     meson_system="linux"
-    target_platform="wayland gbm"
```
Comment (lrusak) [inline tools/depends/configure.ac:385]: The problem here is that we can't build for x11 on the arm/aarch64 builds. We don't cross compile the x11 libs and that's why it was left out here.

So it's still gated in the jenkins Linux build config but it's based on the target arch (like here).

In the depends/target Makefile we build dependencies based on gl/gles.

We should probably be consistent here but it's probably better to use target_platform to determine what dependencies to build.

Comment (fuzzard) [reply tools/depends/configure.ac:385]: That's definitely a path to go as well. Either way is fine to me, so I'll let you Linux folks work out which to take

Comment (wsnipex) [reply tools/depends/configure.ac:385]: Agreed, it makes much more sense to decide depends based on target_platform then on render system

Comment (fuzzard) [reply tools/depends/configure.ac:385]: Ok, i'll start again. The assumption is we want to build dependencies based on target_platform.

Ive pushed a commit (and reverting the original). Absolute stab in the dark about what deps are x11/wayland, and ive no idea if theres anything that would be gbm specific. Pointers for what else you would need appreciated.

**Review by wsnipex — APPROVED** [review]
looks good to me

**Review by lrusak — APPROVED** [review]
I think I'm ok with this. It just breaks up the depends to be more logical based on the target_platform. We hard code the target_platform currently so it's ok.

If in the future we ever make this configurable with the use_platform configure switch we may be in a world of pain having to ifdef the package builds based on target_platform.

**Pull request merged**